### PR TITLE
[MSV1_0] Partly implement network login (part 1-5)

### PR DESCRIPTION
**Pull request by andy-123**
This PR is part of sspi-bringup.

It implements (partly) network login. No (ntlm-)calculations will done so far. This part is left out and will follow any time.
However code is already working in sspi-bringup.

### CODE REVIEW

File: dll/win32/msv1_0/sam.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    //PMSV1_0_LM20_LOGON LogonInfo;
+    //PUNICODE_STRING ComputerName;
+    /* Result (IsNetwork = TRUE) */
+    //LSA_SAM_NETLOGON_TYP LogonType;
+    //LANMAN_SESSION_KEY LanmanSessionKey;
+    //USER_SESSION_KEY UserSessionKey;
+} LSA_SAM_PWD_DATA, *PLSA_SAM_PWD_DATA;
+
+/**
+ * @brief Validates a user by ...
```
Comment (ThFabba) [inline dll/win32/msv1_0/sam.h]: Should there be more here? ;)

File: dll/win32/msv1_0/sam.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ *        Returns the AccontDomainSid
+ * @param PSID
```
Comment (ThFabba) [inline dll/win32/msv1_0/sam.h]: None of the parameters is a PSID ;)

File: dll/win32/msv1_0/msv1_0.c [hunk 3]
```diff
@@ -1179,19 +955,13 @@ LsaApLogonUserEx2(IN PLSA_CLIENT_REQUEST ClientRequest,
...
     UNICODE_STRING ComputerName;
     WCHAR ComputerNameData[MAX_COMPUTERNAME_LENGTH + 1];
-    SAMPR_HANDLE ServerHandle = NULL;
-    SAMPR_HANDLE DomainHandle = NULL;
+    LSA_SAM_PWD_DATA LogonPwdData = { FALSE, NULL };
     SAMPR_HANDLE UserHandle = NULL;
     PRPC_SID AccountDomainSid = NULL;
-    RPC_UNICODE_STRING Names[1];
     SAMPR_ULONG_ARRAY RelativeIds = {0, NULL};
     SAMPR_ULONG_ARRAY Use = {0, NULL};
```
Comment (ThFabba) [inline dll/win32/msv1_0/msv1_0.c]: Looks like you can get rid of these two as well.

File: dll/win32/msv1_0/sam.c [hunk 4]
```diff
@@ -454,3 +454,126 @@ SamValidateNormalUser(
...
+            LogonUserName,
+            LogonPwdData,
+            ComputerName,
+            AccountDomainSidPtr,
+            UserHandlePtr,
+            UserInfoPtr,
+            SubStatus);
+        if (!NT_SUCCESS(Status))
+        {
+            ERR("SamValidateUser() failed (Status 0x%08lx)\n", Status);
```
Comment (ThFabba) [inline dll/win32/msv1_0/sam.c:573]: Should this say `SamValidateNormalUser() failed`?

File: dll/win32/msv1_0/sam.c [hunk 5]
```diff
@@ -454,3 +454,126 @@ SamValidateNormalUser(
...
+                                       HEAP_ZERO_MEMORY,
+                                       sizeof(SAMPR_USER_ALL_INFORMATION));
+            if (*UserInfoPtr == NULL)
+            {
+                Status = STATUS_INSUFFICIENT_RESOURCES;
+                goto done;
+            }
+
+            (*UserInfoPtr)->All.UserId = SECURITY_LOCAL_SERVICE_RID;
+            (*UserInfoPtr)->All.PrimaryGroupId = SECURITY_LOCAL_SERVICE_RID;
```
Comment (ThFabba) [inline dll/win32/msv1_0/sam.c:533]: These success cases should set `UserHandlePtr` and `SubStatus` (or else they need to be marked with something other than an unconditional `_Out_`)

File: dll/win32/msv1_0/sam.h [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ */
+#ifndef _MSV1_0_SAM_H_
+#define _MSV1_0_SAM_H_
+
+typedef enum _LSA_SAM_NETLOGON_TYP
+{
+    NetLogonAnonymouse = 0,
+    NetLogonLmKey,
+    NetLogonNtKey
+} LSA_SAM_NETLOGON_TYP;
```
[the remark this answers is not in the document]
Comment (andy-123) [reply dll/win32/msv1_0/sam.h]: fixed.

File: dll/win32/msv1_0/sam.h (hunk 1 above)
Comment (andy-123) [reply dll/win32/msv1_0/sam.h]: Fixed comment.

File: dll/win32/msv1_0/msv1_0.c (hunk 3 above)
Comment (andy-123) [reply dll/win32/msv1_0/msv1_0.c]: Removed. Thank you.

File: dll/win32/msv1_0/sam.c (hunk 4 above)
Comment (andy-123) [reply dll/win32/msv1_0/sam.c]: Fixed typo. Thanks.

**Pull request merged**